src/client.rs

```diff
@@ -1183,4 +1480,301 @@ mod tests {
+            // server metadata
+            TestResponse {
+                request_url: fake_server.to_string(),
+                response_status: 200,
+                response_body: json!({
+                    "capabilities": {
+                        "attachments": {
+                            "base_url": format!("{}/attachments", fake_server),
+                        }
+                    }
+                }).to_string().as_bytes().to_vec(),
+                response_headers: Headers::new(),
+            },
+
+            // list of changed collections
+            TestResponse {
+                request_url: format!(
+                    "{}/buckets/monitor/collections/changes/changeset?_expected=0",
+                    fake_server
+                ),
+                response_status: 200,
+                response_body: json!({
+                    "metadata": {},
+                    "changes": [{
+                        "id": "record-1",
+                        "last_modified": 0,
+                        "bucket": "main",
+                        "collection": "some-attachments"
+                    }],
+                    "timestamp": 0,
+                }).to_string().as_bytes().to_vec(),
+                response_headers: Headers::new(),
+            },
+
+            // changes for this collection
+            TestResponse {
+                request_url: format!(
+                    "{}/buckets/main/collections/some-attachments/changeset?_expected=0",
+                    fake_server
+                ),
+                response_status: 200,
+                response_body: json!({
+                    "metadata": {},
+                    "changes": [{
+                        "id": "record-1",
+                        "last_modified": 0,
+                        "attachment": {
+                            "hash": "f2ca1bb6c7e907d06dafe4687e579fce76b37e4e93b7605022da52e6ccc26fd2",
+                            "size": 5,
+                            "filename": "test-attachment.txt",
+                            "location": "/1",
+                            "mimetype": "text/plain",
+                        },
+                    }],
+                    "timestamp": 0,
+                }).to_string().as_bytes().to_vec(),
+                response_headers: Headers::new(),
+            },
+
+            // The attachment
+            TestResponse {
+                request_url: format!(
+                    "{}/attachments/1",
+                    fake_server
+                ),
+                response_status: 200,
+                response_body: "test\n".as_bytes().to_vec(),
+                response_headers: Headers::new(),
+            },
+        ];
+
+        let test_client: Box<dyn Requester + 'static> =
+            Box::new(TestHttpClient::new(test_responses));
+
+        let mut client = Client::builder()
+            .server_url(fake_server)
+            .http_client(test_client)
```

Comment:
I would set an explicit verifier here, otherwise the `DummyVerifier` won't verify your hash below